pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -463,4 +491,134 @@ private long getWatermarkMs(long minStartTimeMs, long bucketMs, String mergeLeve
 
     return pinotTaskConfigs;
   }
+
+  private long getMergeRollupTaskDelayInNumTimeBuckets(long watermarkMs, long bufferTimeMs, long bucketTimeMs) {
+    if (watermarkMs == -1) {
+      return 0;
+    }
+    return (System.currentTimeMillis() - watermarkMs - bufferTimeMs) / bucketTimeMs;
+  }
+
+  private void setWatermarkMs(String tableNameWithType, String mergeLevel, long watermarkMs, long bufferTimeMs,
```

Comment:
Better naming? This function updates the watermark map for emitting metrics purposes, we also set the watermark in zk metadata outside of this function. We can either put `mergeRollupTaskMetadata.getWatermarkMap().put(mergeLevel, windowStartMs);` inside this function or rename this function.

Reply:
I renamed the function as `setUpDelayMetric` & `resetDelayMetric`. 

We just cache the watermark locally for emitting the metrics. So, this is actually not so related with bumping up the watermark.